src/Map.cpp

```diff
@@ -9,30 +9,33 @@
 #include <boost/tokenizer.hpp>
 #include <boost/algorithm/string.hpp>
 #include <vector>
+#include <boost/filesystem.hpp>
 #include "Map.hpp"
 
-// Initialize member variables, reserve buffer space, and open the output file
-Map::Map(char* outFile, size_t bufSize){
+// Initialize FileManager instance and reserve buffer space
+Map::Map(std::string inputFile, size_t bufSize, size_t lineCount){
+    this->fileManager = FileManager::GetInstance();
+    tempFilename = inputFile;
     bufferSize = bufSize;
     buffer.reserve(bufferSize);
-    outFilename = outFile;
-    outputFile.open(outFilename);
+    numLines = lineCount;
```

Comment:
This might be my security side coming out, but lineCount shouldn't be something that is user-specified and left unvalidated. There are two options here:

1. We can either have a caller object (Workflow, for example) parse the lines and give it to the Map object (though that wouldn't actually reflect the instructions for Phase 1); or
2. Map should figure out the line count by looping through the file line by line and keeping a count of how many lines it's seen until it reaches EOF. 

In the case of (2), it might be worth keeping a count of how many words had been tokenized too.

Reply:
LineCount should be populated in workflow based on the parsing of the input file that was done in File Manager, so it seems that we're currently doing (1) above. I can change to (2) by instead incrementing the member variable with each call to map(), this was it won't have to be passed into the constructor.